src/Exceptions/Interfaces/ExceptionInterface.php

```diff
@@ -52,6 +52,12 @@
         'title' => 'Unprocessable entity',
     ];
 
+    // Billing/payment related errors 11000 - 11999
+    const PAYMENT_REQUIRED = [
```

Comment:
Is there a reason you're setting this constant on the interface, rather than on the actual class you're using it?

Reply:
We're putting all these constants with their error codes etc on the interface to have them in 1 place, so we don't accidentally duplicate error codes and have an overview of all available errors.

Reply:
This way it's organized and easy to distinguish between the different available error codes without having to look in other classes.

Reply:
Hm okay, I'd rather have an `abstractJsonApiException` where these are defined instead of defining constants on an interface. Feels like a bit of an anti-pattern to me, but keep it for now if this is the way it's been done.

Reply:
No, using an abstract class for that would be an anti-pattern. It is the interface's responsibility to make sure all classes implementing it are consistent. A class may implement this interface, but not extend the abstract class. Which would mean it would mean there is no more indication as to what error codes etc should be used.

Reply:
I don't see how defining all constants on the interface makes its implementation any more or less consistent. If anything, it makes the implementations heavier than they have to be. Why should **an** exception contain data about **all** possible errors when it eventually just uses one. Essentially this also changes all exception classes any time an error is added. Would it be an idea to create a dedicated, enum-like, class to contain all the possible errors, as the list off errors is prone to change often?

Reply:
Sure, an enum like class could be preferred to using an interface. But using an interface for these kind of things is not uncommon or an anti-pattern.

Reply:
Anyway, changing that now would be far out of the scope of this issue, so if this is the only issue you have left with this PR, please approve it, so Nick can continue :)

Reply:
Aye, that's fair enough